apps/docs/components/DocSearch.tsx

```diff
@@ -115,6 +115,10 @@ export function SearchProvider({ children }: any) {
   )
 }
 
+const SearchProvider = memo(SearchProviderContent)
```

Comment:
This memo is probably adding more overhead than it's saving. The `children` prop will probably change on every render, so this memo isn't effective.

Adding a memo around whatever you passing in as `children` will probably yield better results. For example:

```typescript
<SearchProvider>
  <MyComp /> // MyComp should be memo'd
</SearchProvider>
```

This way, when the state changes in the provider, MyComp won't rerender as its props don't change.

Reply:
oh yea, this was from back when i was memoing everything, as i was trying to find what was the bottleneck.
I'll remove.

turns out the biggest offender was next router 🙄

Reply:
removed